impersonate/middleware.py

```diff
@@ -1,26 +1,55 @@
+from django.utils.functional import empty, SimpleLazyObject
 from .helpers import User, check_allow_for_uri, check_allow_for_user
 
 
-class ImpersonateMiddleware(object):
+def apply_impersonate(request):
+    request.user.is_impersonate = False
+    request.__dict__['impersonator'] = None
```

Comment:
oh, right. Going through __dict__ is probably not necessary for request. It will be for the other stuff thoguh

https://github.com/django/django/blob/stable/1.8.x/django/utils/functional.py#L248